frontend/src/Pages/NewKeepitPage.js

```diff
@@ -0,0 +1,52 @@
+import { useHistory } from 'react-router-dom'
+import { useState, useEffect } from 'react'
+import UploadButton from '../Components/UploadButton'
+
+export default function NewKeepitPage() {
+  const history = useHistory()
+  const [images, setImages] = useState()
+  const [showUploadButton, setShowUploadButton] = useState(false)
+
+  useEffect(() => {
+    if (images && images.length > 0) {
+      setShowUploadButton(false)
+    } else {
+      setShowUploadButton(true)
+    }
```

Comment:
You can simplify this:

```suggestion
    setShowUploadButton(images && images.length > 0))   
```

or (more magic going on here 🎉)

```suggestion
    setShowUploadButton(Boolean(images?.length))   
```